packages/ember-htmlbars/tests/helpers/component_test.js

```diff
@@ -1,398 +0,0 @@
-    location: 'Caracas',
-    template: compile('{{#component view.dynamicComponent location=view.location}}arepas!{{/component}}')
-  });
-
-  runAppend(view);
-  equal(view.$().text(), 'yippie! Caracas arepas!', 'component was looked up and rendered');
-
-  run(function() {
-    set(view, 'dynamicComponent', 'baz-qux');
-    set(view, 'location', 'Loisaida');
-  });
-  equal(view.$().text(), 'yummy Loisaida arepas!', 'component was updated and re-rendered');
-});
-
-QUnit.test('component helper destroys underlying component when it is swapped out', function() {
-  var currentComponent;
-  var destroyCalls = 0;
-  owner.register('component:foo-bar', Component.extend({
-    init() {
-      this._super(...arguments);
-      currentComponent = 'foo-bar';
-    },
-    willDestroy() {
-      destroyCalls++;
-    }
-  }));
-  owner.register('component:baz-qux', Component.extend({
-    init() {
-      this._super(...arguments);
-      currentComponent = 'baz-qux';
-    },
-    willDestroy() {
-      destroyCalls++;
-    }
-  }));
-
-  view = EmberView.create({
-    [OWNER]: owner,
-    dynamicComponent: 'foo-bar',
-    template: compile('{{component view.dynamicComponent}}')
-  });
-
-  runAppend(view);
-
-  equal(currentComponent, 'foo-bar', 'precond - instantiates the proper component');
-  equal(destroyCalls, 0, 'precond - nothing destroyed yet');
-
-  run(function() {
-    set(view, 'dynamicComponent', 'baz-qux');
-  });
-
-  equal(currentComponent, 'baz-qux', 'changing bound value instantiates the proper component');
-  equal(destroyCalls, 1, 'prior component should be destroyed');
-
-  run(function() {
-    set(view, 'dynamicComponent', 'foo-bar');
-  });
-
-  equal(currentComponent, 'foo-bar', 'changing bound value instantiates the proper component');
-  equal(destroyCalls, 2, 'prior components destroyed');
-});
-
-QUnit.test('component helper with actions', function() {
-  owner.register('template:components/foo-bar', compile('yippie! {{yield}}'));
-  owner.register('component:foo-bar', Component.extend({
-    classNames: 'foo-bar',
-    didInsertElement() {
-      // trigger action on click in absence of app's EventDispatcher
-      var self = this;
-      this.$().on('click', function() {
-        self.sendAction('fooBarred');
-      });
-    },
-    willDestroyElement() {
-      this.$().off('click');
-    }
-  }));
-
-  var actionTriggered = 0;
-  var controller = Controller.extend({
-    dynamicComponent: 'foo-bar',
-    actions: {
-      mappedAction() {
-        actionTriggered++;
-      }
-    }
-  }).create();
-  view = EmberView.create({
-    [OWNER]: owner,
-    controller: controller,
-    template: compile('{{#component dynamicComponent fooBarred="mappedAction"}}arepas!{{/component}}')
-  });
-
-  runAppend(view);
-  run(function() {
-    view.$('.foo-bar').trigger('click');
-  });
-  equal(actionTriggered, 1, 'action was triggered');
-});
-
-QUnit.test('component helper maintains expected logical parentView', function() {
```

Comment:
deleted as it's covered by the existing `@test it has the right parentView and childViews`